models/users.js

```diff
@@ -83,6 +84,42 @@ const getJoinData = async (userId) => {
   }
 };
 
+/**
+ * Fetches users with the given skill
+ *
+ * @param skill { string }: Skill
+ * @return @return {Promise<users>}
+ */
+
+const getSuggestedUsers = async (skill) => {
+  try {
+    const data = await itemModel.where("itemtype", "==", "USER").where("tagid", "==", skill).get();
+    const usersId = [];
+    const users = [];
+
+    if (!data.empty) {
+      data.forEach((doc) => {
+        usersId.push(doc.data().itemid);
+      });
+      const usersArray = usersId.map((userId) => fetchUser({ userId }));
+      const userDuplicate = await Promise.all(usersArray);
```

Comment:
I am not sure if the naming here looks right. What does `userDuplicate` signify? It looks like the `usersArray` is just used to get back a list of users' data.